src/ir_Fujitsu.cpp

```diff
@@ -0,0 +1,218 @@
+//   repeat: Nr. of times the message is to be repeated.
+//          (Default = FUJITSU_AC_MIN_REPEAT).
+//
+// Status: BETA / Appears to be working.
+//
+void IRsend::sendFujitsuAC(unsigned char data[], uint16_t nbytes,
+                           uint16_t repeat) {
+  // Set IR carrier frequency
+  enableIROut(38);
+  for (uint16_t r = 0; r <= repeat; ++r) {
+    // Header
+    mark(FUJITSU_AC_HDR_MARK);
+    space(FUJITSU_AC_HDR_SPACE);
+    // Data
+    for (uint16_t i = 0; i < nbytes; i++)
+    sendData(FUJITSU_AC_BIT_MARK, FUJITSU_AC_ONE_SPACE,
+         FUJITSU_AC_BIT_MARK, FUJITSU_AC_ZERO_SPACE,
+         data[i], 8, false);
+    // Footer
+    mark(FUJITSU_AC_TRL_MARK);
+    space(FUJITSU_AC_TRL_SPACE);
+  }
+}
+
+// Code to emulate Fujitsu A/C IR remote control unit.
+
+// Warning: Consider this very alpha code. Seems to work, but not validated.
+//
+// Equipment it seems compatible with:
+//  * Fujitsu ASYG30LFCA with remote AR-RAH2E
+//  * <Add models (A/C & remotes) you've gotten it working with here>
+// Initialise the object.
+IRFujitsuAC::IRFujitsuAC(uint16_t pin) : _irsend(pin) {
+  stateReset();
+}
+
+// Reset the state of the remote to a known good state/sequence.
+void IRFujitsuAC::stateReset() {
+  _temp = 24;
+  _fanSpeed = FUJITSU_AC_FAN_HIGH;
+  _mode = FUJITSU_AC_MODE_COOL;
+  _swingMode = FUJITSU_AC_SWING_BOTH;
+  _cmd = FUJITSU_AC_CMD_TURN_ON;
+}
+
+// Configure the pin for output.
+void IRFujitsuAC::begin() {
+    _irsend.begin();
+}
+
+// Send the current desired state to the IR LED.
+void IRFujitsuAC::send() {
+  getRaw();
+  uint8_t len = getCommandLength();
+  _irsend.sendFujitsuAC(remote_state, len);
+}
+
+uint8_t IRFujitsuAC::getCommandLength() {
+  uint8_t len = FUJITSU_AC_STATE_LENGTH;
+  if (remote_state[5] != 0xFE)
+    len = FUJITSU_AC_STATE_LENGTH_SHORT;
+  return len;
+}
+
+// Return a pointer to the internal state date of the remote.
+uint8_t* IRFujitsuAC::getRaw() {
+  remote_state[0] = 0x14;
+  remote_state[1] = 0x63;
+  remote_state[2] = 0x00;
+  remote_state[3] = 0x10;
+  remote_state[4] = 0x10;
+  bool fullCmd = false;
+  switch (_cmd) {
+    case FUJITSU_AC_CMD_TURN_OFF:
+      remote_state[5] = 0x02;
+      break;
+    case FUJITSU_AC_CMD_STEP_HORIZ:
+      remote_state[5] = 0x79;
+      break;
+    case FUJITSU_AC_CMD_STEP_VERT:
+      remote_state[5] = 0x6C;
+      break;
+    default:
+      remote_state[5] = 0xFE;
+      fullCmd = true;
+      break;
+  }
+  if (fullCmd) {
+    remote_state[6] = 0x09;
+    remote_state[7] = 0x30;
+    uint8_t tempByte = _temp - 16;
+    remote_state[8] = (_cmd == FUJITSU_AC_CMD_TURN_ON) | (tempByte << 4);
+    remote_state[9] = _mode | 0 << 4;  // timer off
+    remote_state[10] = _fanSpeed | _swingMode << 4;
+    remote_state[11] = 0;  // timerOff values
+    remote_state[12] = 0;  // timerOff/on values
+    remote_state[13] = 0;  // timerOn values
+    remote_state[14] = 0x20;
+    // Checksum is such that all bytes 8 to 16 (NB 16 is the checksum itself)
```

Comment:
The comment doesn't make sense. i.e. It reads like the checksum is included in the checksum'ed range. Which would lead to very non-deterministic checksums. ;-)
That was the point/suggestion of the first comment on this comment.
Your reply makes a good comment. Include that instead.